jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java

```diff
@@ -673,7 +669,7 @@ void stop()
         /**
          * Stores our last MUC presence packet for future update.
          */
-        private Presence lastPresenceSent;
+        private PresenceBuilder lastPresenceSent;
```

Comment:
In Smack, Stanza modification is synchronized, but builders are not.  I haven't analyzed the threading model here, but is there any danger with this?

Reply:
Hm, good question. What do we actually need/want to keep from the presence packet, i.e. why is it intercepted and stored instead of always built from scratch?

I could synchronize on the builder while modifying the extension list and while calling .build(), but that's ugly :-/

Reply:
I'm really not sure who calls what here. There's some synchronization from MucClient and MucClientManager, but that's unlikely to be meant and usable for the presence object. I've added some synchronization that should be cheap enough to and not interfere with Smack.